platforms/cpuinfo.go

```diff
@@ -98,7 +101,11 @@ func getCPUVariant() string {
 	variant, err := getCPUInfo("Cpu architecture")
 	if err != nil {
 		log.L.WithError(err).Error("failure getting variant")
-		return ""
+		variant, err = getCPUArchitecture()
+		if err != nil {
+			log.L.WithError(err).Error("failure getting variant with getCPUArchitecture")
```

Comment:
It's a bit odd to unconditionally log an error here; perhaps;

```suggestion
			if !errdefs.IsNotImplemented(err) {
				log.L.WithError(err).Error("failure getting variant with getCPUArchitecture")
			}
```

(if we don't implement (or can't support) it on specific platforms, we could also consider creating a stub in platform-specific files).